checkov/terraform/runner.py

```diff
@@ -87,21 +94,23 @@ def check_tf_definition(self, report, root_folder, runner_filter, collect_skip_c
             definitions_context = {}
             for definition in self.tf_definitions.items():
                 definitions_context = parser_registry.enrich_definitions_context(definition, collect_skip_comments)
-            variable_evaluator = ConstVariableEvaluation(root_folder, self.tf_definitions, definitions_context)
-            variable_evaluator.evaluate_variables()
-            self.tf_definitions, self.definitions_context = variable_evaluator.tf_definitions, variable_evaluator.definitions_context
```

Comment:
@robeden The `evaluate_variables()` method also turned all hardocded "true" and "false" in resources to True / False. Is that taken care of in your new parser?

Reply:
I think `true` and `false` are handled, but let me make sure there's a test case to be sure. It also converted `1` and `0`, which I'm less excited about converting.

I have a plan for getting back to TF 0.11 support and that location might be a better place to special-case that if we want, since I believe it's behavior specific to <0.12?

Reply:
Looks like it's not currently handled. I threw in the easy implementation, but that broke some more sophisticated logic in the `tomap_function` parser scenario (specifically the MIXED_BOOL) test case which is ensuring the datatype in maps is consistent. I'll have to think on how to prevent that. (If it matters. For checkov's purposes, some portions of the spec probably aren't important.)

Reply:
I agree that `false` and `true` strings are more important to handle. Can you handle them?

Reply:
Yep, will do.

Reply:
What do you think about converting “0” and “1”? How big a deal are they?

Reply:
`true` and `false` are now handled. I haven't done anything with `1` and `0` for now.

Reply:
0 and 1 are old conventions, and prone to many false positives as they are valid values by themselves. I'd leave it as is for now